public/js/main.js

```diff
@@ -189,54 +173,85 @@ function setFileURLs(new_provider) {
+    if (!(/iPhone|iPad/i.test(navigator.userAgent))) {
+      $('.packages-table-container table tbody tr, .library-table-container table tbody tr').hover(function (ev) {
+        // If we're on a search page, update the buttons for the right lib
+        if ($(".packages-table-container")[0]) {
+          copyEl.children('ul').remove();
+          copyEl.append(generateCopyDropdown(this.dataset.sri));
+        }
+
+        // Show the button
+        copyEl.appendTo($(ev.currentTarget).find('.library-column'));
+        copyEl.show();
+      }, function () {
+        // Hide the button when hover ends
+        copyEl.hide();
+      });
+
+      // Setup the clipboard handler
+      if (clipboard) clipboard.destroy();
+      setupCopyButton();
+    }
+  }
+
+  /****
+   * Search Scripting
+   ****/
+
+  var displayPage = 0,
+    queryItems = 20,
+    cachedQueryResult = {},
+    lazyScroll = false,
+    $nbHitsField = $('#nb-hits-field'),
+    $processingTimeMS = $('#processingTimeMS'),
+    $hits = $('.packages-table-container tbody'),
+    $search = $('#search-box'),
+    lastHashQuery,
+    algolia = algoliasearch('2QWLVLXZB6', '2663c73014d2e4d6d1778cc8ad9fd010'), // public/search-only credentials
+    index = algolia.initIndex('libraries'),
+    lastQuery;
+
+  function getSafeHighlightedValue(highlight) {
+    // Extract & escape the attribute to prevent any XSS issue keeping the highlighting tags
+    return $('<div />').text(highlight && highlight.value || '').html().replace(/&lt;(\/?)em&gt;/g, '<$1em>');
+  }
 
-  var displayPage = 0;
-  var queryItems = 20;
-  var cachedQueryResult = {};
-  var lazyScroll = false;
-  var $nbHitsField = $('#nb-hits-field');
-  var $processingTimeMS = $('#processingTimeMS');
-  var $hits = $('.packages-table-container tbody');
-  var $allRows = $hits.html();
   function displayMatchingLibraries(err, content) {
     $('.packages-table-container').show();
-    if (err) {
-      appLoading.stop();
-    }
 
-    if (err || content.query !== $('#search-box').val()) {
+    // If we had an error, or the results don't match the input, abort
+    if (err || content.query !== $search.val()) {
+      appLoading.stop();
       return;
     }
 
-    function getSafeHighlightedValue(highlight) {
-      // extract & escape the attribute to prevent any XSS issue keeping the highlighting tags
-      var v = highlight && highlight.value || '';
-      return $('<div />').text(v).html().replace(/&lt;(\/?)em&gt;/g, '<$1em>');
-    }
-
+    // If we've got all the results, stop searching on scroll
     if ((displayPage + 1) * queryItems >= content.nbHits) {
       lazyScroll = false;
     }
 
-    // set total hits found, if not set to 0.
+    // Set total hits found, if not set to 0.
     $nbHitsField.text(content.nbHits || 0);
     $processingTimeMS.text(content.processingTimeMS || 0);
 
-    var html = '';
-    var match = false;
-    if (content.hits.length < 1) {
+    // If there are no results, hide the table
```

Comment:
Minor: Comments are generally for explaining why you did something or to give additional context to something. This comment is just a line 241 in english.

Reply:
Agreed, generally, it'd be a silly thing to comment, but I think I'd like to have as much of the logic as possible commented in this file to encourage any future editors to heavily comment their code so that we know what's going on.

Thoughts?